compiler/rustc_mir_build/src/check_unsafety.rs

```diff
@@ -90,23 +91,50 @@ impl<'tcx> UnsafetyVisitor<'_, 'tcx> {
     }
 
     fn emit_deprecated_safe_fn_call(&self, span: Span, kind: &UnsafeOpKind) -> bool {
+        fn parse_rustc_deprecated_safe_2024_attr(attr: &Attribute) -> Option<Symbol> {
+            for item in attr.meta_item_list().unwrap_or_default() {
+                if item.has_name(sym::audit_that) {
+                    return Some(item.value_str().expect(
```

Comment:
```suggestion
                    if let Some(msg) = item.value_str() {
```
The current code  will most likely ICE if the value is not actually string.

Reply:
Not a good practice, but not a blocker.

Reply:
I have verified that this indeed ICEs. I'd still prefer the ICE over silently ignoring the attribute value. The compiler error says:
```
error: the compiler unexpectedly panicked. this is a bug.

note: using internal features is not supported and expected to cause internal compiler errors when used incorrectly
```